astroquery/mast/observations.py

```diff
@@ -642,8 +648,8 @@ def _download_curl_script(self, products, out_dir):
         """
 
         url_list = [("uri", url) for url in products['dataURI']]
-        download_file = "mastDownload_" + time.strftime("%Y%m%d%H%M%S")
-        local_path = os.path.join(out_dir.rstrip('/'), download_file + ".sh")
```

Comment:
The `.rstrip("/")` here is unnecessary and is actually a bug if the path is a `Path` object, something writing the test caught.